### PR TITLE
search: google like search fields

### PR DESCRIPTION
**Pull request by Panos512**
* Searches on different fields based on the index we are searching on.

Signed-off-by: Panos Paparrigopoulos <panos.paparrigopoulos@cern.ch>
Co-authored-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

File: inspirehep/modules/search/query_factory.py [hunk 1]
```diff
@@ -44,10 +45,40 @@
...
+    else:
+        return ['global_fulltext']
+
+
 def inspire_query_factory():
     """Create a parser returning Elastic Search DSL query instance."""
 
-    def invenio_query(pattern):
+    def invenio_query(pattern, index='records-hep'):
         walkers = [PypegConverter(), SpiresToInvenio()]
```
Comment (Panos512) [inline inspirehep/modules/search/query_factory.py]: @kaplun @jmartinm should I move this on top?

**Comment by jmartinm** [comment]
By the way, in case it is useful, this is the config of current production regarding `global index`

screen shot 2016-05-18 at 1 27 35 pm

So the work there would be to check which fields correspond to the MARC tags indicated in the screenshot (looking at our [dojson config for each collection](https://github.com/inspirehep/inspire-next/tree/master/inspirehep/dojson))

**Comment by kaplun** [comment]
If I am not mistaken, in global we index everything but fulltext files. There is a white list there just not to index erroneous fields.

File: inspirehep/modules/search/query_factory.py (hunk 1 above)
Comment (jmartinm) [inline inspirehep/modules/search/query_factory.py]: Sure, just double check that everything works as expected after the move ;)

**Comment by annetteholtkamp** [comment]
Is it really everything? Then it's different behaviour to prod. Is that intended?

- Annette


On May 18, 2016, at 4:38 AM, Samuele Kaplun <notifications@github.com<mailto:notifications@github.com>> wrote:


If I am not mistaken, in global we index everything but fulltext files. There is a white list there just not to index erroneous fields.

**Comment by kaplun** [comment]
@annetteholtkamp what do you mean? Is what different from what? I just say that on prod it seems we basically index everything. But indeed being a white list we might miss some important subfield...

**Comment by kaplun** [comment]
But yest, in Labs we want a different behaviour to prod. In particular we decided in the past to try going for google like results (i.e. default terms in OR) and boost certain fields more.

**Comment by annetteholtkamp** [comment]
E.g. the recid is not in the global index. Neither is the collection in 980.

- Annette

On May 21, 2016, at 11:13 AM, Samuele Kaplun <notifications@github.com<mailto:notifications@github.com>> wrote:


@annetteholtkamp<https://github.com/annetteholtkamp> what do you mean? Is what different from what? I just say that on prod it seems we basically index everything. But indeed being a white list we might miss some important subfield...

**Comment by kaplun** [comment]
@Panos512 can you rebase one last time? I think we can merge this even if not perfect. At least we have a base to iterate on top for specifying fields.

**Comment by kaplun** [comment]
BTW this superseded #1098, correct?

**Comment by Panos512** [comment]
@kaplun just rebased. Yes it superseeds it.

**Pull request merged**